app/forms/hub/create_client_form.rb

```diff
@@ -52,6 +55,16 @@ class CreateClientForm < ClientForm
     validate :at_least_one_tax_return_present
     validates :state_of_residence, inclusion: { in: States.keys }
     validates :preferred_interview_language, presence: true, allow_blank: false
+    validates :primary_tin_type, presence: true
+    validates_confirmation_of :primary_ssn
+    validates_presence_of :primary_ssn_confirmation, if: :primary_ssn
+    validates_presence_of :spouse_ssn_confirmation, if: :spouse_ssn
+    validates :primary_ssn, social_security_number: true, if: -> { ["ssn", "ssn_no_employment"].include? primary_tin_type }
+    validates :primary_ssn, individual_taxpayer_identification_number: true, if: -> { primary_tin_type == "itin"}
```

Comment:
Can/should we only allow these three tin types for primary SSN? I ask b/c as-is, it looks like if they enter ATIN as the tin type, they can skip all validation, which just seems bad.

Reply:
ATIN is not an option on the dropdown for primary and spouse so I think the implementation (which is copy and pasted from the other implementations) doesn't account for that but we could make sure that primary and spouse tin_types must be within the set of approved types?

For in intake steps, I noticed that we were doing this "if its this if its that" without validating that tin_type was present, which shouldnt really be possible based on how the form was working but I think it was somehow causing some of the primary_snn.last_4 bugs we saw. I have added validates_presence_of :primary_tin_type in all places for that reason